File: 0129-求根到叶子节点数字之和/Solution.java
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

class Solution {
    public int sumNumbers(TreeNode root) {
        if(root == null){ return 0;}
        if(root.left == null && root.right == null){ return root.val; }
        return helper(root.left, root.val) + helper(root.right, root.val);
    }

    private int helper(TreeNode root, int sum){
        if(root == null){ return 0;}
        int nsum = sum*10 + root.val;
        if (root.left == null && root.right == null) {
            return nsum;
        }
        int lsum = helper(root.left, nsum);
        int rsum = helper(root.right, nsum);
        return lsum + rsum;
    }
}

File: 0129-求根到叶子节点数字之和/solution.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def __init__(self):
        self.ret = 0

    def sumNumbers(self, root: TreeNode) -> int:
        self.caculate(root, 0)
        return self.ret

    def caculate(self, root, prevSum):
        if root is None:
            return None
        tmp = prevSum * 10 + root.val
        if root.left is None and root.right is None:
            self.ret += tmp
        self.caculate(root.left, tmp)
        self.caculate(root.right, tmp)
        